### PR TITLE
Use custom generator

### PR DESCRIPTION
**Pull request by therve**
This replaces openapi-generator by a custom generator. Most of the changes are around docstrings, though it fixes one bug with oneOf of basic types, and cleans up imports of oneOf models.

### CODE REVIEW

File: .generator/src/generator/templates/exceptions.j2 [hunk 1]
```diff
@@ -104,11 +103,9 @@ class ApiException(OpenApiException):
 
     def __str__(self):
         """Custom error messages for exception"""
-        error_message = "({0})\n"\
-                        "Reason: {1}\n".format(self.status, self.reason)
+        error_message = "({0})\n" "Reason: {1}\n".format(self.status, self.reason)
```
Comment (jirikuncar) [inline .generator/src/generator/templates/exceptions.j2:106]:
```suggestion
        error_message = "({0})\nReason: {1}\n".format(self.status, self.reason)
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -28,15 +22,14 @@ repos:
...
       - Sphinx
-  - id: openapi-generator
-    name: OpenAPI-generator
-    language: node
-    entry: bash -c "cd .generator && openapi-generator-cli generate"
+  - id: generator
+    name: generator
+    language: python
+    language_version: "3.9"
+    entry: bash -c "cd .generator && python -m pip install . && python -m generator -i ./schemas/v1/openapi.yaml -o ../src/ && python -m generator -i schemas/v2/openapi.yaml -o ../src/"
```
Comment (jirikuncar) [inline .pre-commit-config.yaml:29]: I would split it so we don't need to regenerate both v1 and v2 every time.

File: .generator/src/generator/templates/api.j2 [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    file_type,
+    none_type,
+)
+{%- for model in get_api_models(operations) %}
+from {{ config["packageName"] }}.model.{{ model|snake_case }} import {{ model }}
+{%- endfor %}
+
+{% set classname = name.replace(" ", "") + "Api" %}
+class {{ classname }}:
+
```
Comment (jirikuncar) [inline .generator/src/generator/templates/api.j2:17]: can we add a tag description as docstring here?

File: .generator/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+[tool.poetry]
+name = "generator"
+version = "0.1.0"
+description = ""
+authors = ["Jiri Kuncar <jiri.kuncar@datadoghq.com>"]
```
Comment (jirikuncar) [inline .generator/pyproject.toml:5]: 😉

File: .generator/src/generator/templates/api.j2 (hunk 3 above)
Comment (therve) [reply .generator/src/generator/templates/api.j2:17]: I added it in the init method, is that good enough?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (therve) [reply .pre-commit-config.yaml:29]: Done.

File: .generator/pyproject.toml (hunk 4 above)
Comment (therve) [reply .generator/pyproject.toml:5]: 😄 . Should we put some generic name like support@?

Comment (jirikuncar) [reply .generator/pyproject.toml:5]: Probably :)

File: .generator/src/generator/templates/configuration.j2 [hunk 5]
```diff
@@ -154,37 +150,36 @@ class Configuration(object):
...
+            "update_slo_correction": False,
+            "get_slo_corrections": False,
+            "get_slo_history": False,
+            "get_daily_custom_reports": False,
+            "get_hourly_usage_attribution": False,
+            "get_monthly_custom_reports": False,
+            "get_monthly_usage_attribution": False,
+            "get_specified_daily_custom_reports": False,
+            "get_specified_monthly_custom_reports": False,
+            "get_usage_attribution": False,
```
Comment (skarimo) [inline .generator/src/generator/templates/configuration.j2:166]: We should dynamically populate this

File: .generator/src/generator/templates/configuration.j2 [hunk 6]
```diff
@@ -421,33 +397,55 @@ class Configuration(object):
...
-                        {{/-last}}
-                        {{/enumValues}}
-                        }{{^-last}},{{/-last}}
-                {{#-last}}
-                    }
-                {{/-last}}
-                {{/variables}}
-            }{{^-last}},{{/-last}}
-            {{/servers}}
+                "url": "https://{subdomain}.{site}",
```
Comment (skarimo) [inline .generator/src/generator/templates/configuration.j2:394]: Same here as above. Any reason its hard coded?

File: .generator/src/generator/templates/package.j2 [hunk 7]
```diff
@@ -0,0 +1,59 @@
+=begin
```
Comment (skarimo) [inline .generator/src/generator/templates/package.j2:1]: Looks like template from ruby. Do we need this file?

File: .generator/src/generator/templates/configuration.j2 (hunk 6 above)
Comment (therve) [reply .generator/src/generator/templates/configuration.j2]: Nope, big mistake.

**Pull request merged**